mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp

```diff
@@ -85,25 +86,20 @@ LogicalResult PopulateParams::calculateBlockGemmPerformanceParameters(
         param.gemmNPerBlock % param.gemmNPerThread == 0))
     return failure();
 
-  int64_t threadGemmMPerBlock = param.gemmMPerBlock / param.gemmMPerThread;
-  int64_t threadGemmNPerBlock = param.gemmNPerBlock / param.gemmNPerThread;
-
-  int64_t threadGemmMPerCluster =
-      derived.mThreadsPerCuwave * derived.mCuwavesPerBlock;
-  int64_t threadGemmNPerCluster =
-      derived.nThreadsPerCuwave * derived.nCuwavesPerBlock;
-
-  if (!(threadGemmMPerBlock % threadGemmMPerCluster == 0) &&
-      (threadGemmNPerBlock % threadGemmNPerCluster == 0))
-    return failure();
-
-  int64_t clusterMPerBlock = threadGemmMPerBlock / threadGemmMPerCluster;
-  int64_t clusterNPerBlock = threadGemmNPerBlock / threadGemmNPerCluster;
-
-  // inline asm only support clusterMPerBlock = 2 andclusterNPerBlock =
-  // 2
-  if (!(clusterMPerBlock == 2 && clusterNPerBlock == 2))
+  int64_t threadGemmMPerCluster = param.gemmMPerThread *
+                                  derived.mThreadsPerCuwave *
+                                  derived.mCuwavesPerBlock;
```

Comment:
Hmm so previously we always assume `gemmMPerThread == 1` and it actually doesn't break anything?

Reply:
No, previously we'd divided by gemmMPerThread before doing the check. There could've been scenarios where it was broken.